pkg/build/util/generate_test.go

```diff
@@ -80,6 +82,152 @@ func TestGenerateBuildFromBuild(t *testing.T) {
 	}
 }
 
+func TestSubstituteImage(t *testing.T) {
+	source := mockSource()
+	strategy := mockDockerStrategy()
+	output := mockOutput()
+	const newImage = "newImage"
+
+	bc := &api.BuildConfig{
+		ObjectMeta: kapi.ObjectMeta{
+			Name: "test-build-config",
+		},
+		Parameters: api.BuildParameters{
+			Source: source,
+			Revision: &api.SourceRevision{
+				Type: api.BuildSourceGit,
+				Git: &api.GitSourceRevision{
+					Commit: "1234",
+				},
+			},
+			Strategy: strategy,
+			Output:   output,
+		},
+	}
+
+	// Docker build with nil base image
+	// base image should still be nil
+	build := GenerateBuildFromConfig(bc, nil)
```

Comment:
I've noticed you used the same pattern in `image_change_controller_test.go`, and I allowed myself to change it (again in https://github.com/bparees/origin/pull/1). My understanding of unit test you should have a clean state between tests, here you're being exposed to changes hiding between subsequent calls to `SubstituteImageReferences`. I'd propose having each of these calls as a separate test method, where you set them up fresh and then proceed with the test.